tests/middleware/test_base.py

```diff
@@ -210,24 +211,17 @@ async def dispatch(
     ) -> Response:
         ctxvar.set("set by middleware")
         resp = await call_next(request)
-        assert ctxvar.get() == "set by endpoint"
+        # BaseHTTPMiddleware creates a TaskGroup which copies the context
+        # and erases any changes to it made within the TaskGroup
+        assert ctxvar.get() == "set by middleware"
```

Comment:
Well, yeah, that's why the `pytest.mark.xfail` is there... 🤔

Reply:
`xfail` complicates testing for a warning, which is why I had to make adjustments.

An `AssertionError` within the middleware triggers the same `ResourceWarning` (refer to issue #2778). However, since the warning is emitted only after the stream is deleted, it appears some time later, specifically after an explicit `gc.collect()` in `test_anyio_streams_cleanup_exc_in_route`. This causes the test to fail due to a leftover stream
from another test.

Perhaps there's no need for this test at all, as it is currently quite brittle. 🤷‍♂️ It might be better to stop ignoring the warning from anyio once all streams are properly closed.